crates/ty_python_semantic/src/types/call/bind.rs

```diff
@@ -972,6 +972,28 @@ impl<'db> Bindings<'db> {
                             }
                         }
 
+                        Some(KnownClass::Tuple) if overload_index == 1 => {
+                            if let [Some(argument)] = overload.parameter_types() {
+                                let overridden_return =
+                                    argument.into_tuple().map(Type::Tuple).unwrap_or_else(|| {
```

Comment:
I think this would avoid unwrapping the argument type just to then rewrap it:

```suggestion
                                    argument.filter(Type::is_tuple).unwrap_or_else(|| {
```

(though it depends on an `is_tuple` method that may not exist yet!)

Reply:
`argument` is a type rather than an `Option` here so I think this would have to be

```suggestion
                                    argument.into_tuple().filter(Type::is_tuple).unwrap_or_else(|| {
```

Adding a `Type::is_tuple` method just for that feels a bit unnecessary to me? No strong opinion though